source/fundamentals/connection.txt

```diff
@@ -0,0 +1,435 @@
+   provides a convenient pattern to guarantee that a ``MongoClient``
+   is always closed.
+
+.. _connection-uri:
+
+Connection String
+--------------
+
+The **connection string** (also known as a **connection uri**) provides
+instructions on how to connect to a MongoDB deployment. The driver
+interprets the connection string to determine where and how
+it should connect to MongoDB as well as how to behave while
+connected. The following example shows each part of the connection URI:
+
+.. figure:: /includes/figures/connection_string_parts.png
+
+In this example, for the protocol, we use ``mongodb+srv`` which specifies the
+:manual:`DNS Seedlist Connection Format
+</reference/connection-string/#dns-seedlist-connection-format>`. This indicates
+that the hostname following it corresponds to the DNS SRV record of your
+MongoDB instance or deployment. If your instance or deployment does not have a
+DNS SRV record, use ``mongodb`` to specify the :manual:`Standard Connection
+Format </reference/connection-string/#standard-connection-string-format>`.
+
+.. note::
+
+   If your deployment is on MongoDB Atlas, follow the
+   :atlas:`Atlas driver connection guide <driver-connection?jmp=docs_driver_java>`
+   to retrieve your connection string.
+
+The next part of the connection string contains your username and password
+if you are using password-based authentication. Replace the value of ``user``
+with your username and ``pass`` with your password. If you are using an
+authentication mechanism that does not require a username and password, omit
+this part of the connection URI.
+
+The next part of the connection string specifies the hostname or IP address and
+port of your MongoDB instance. In the example below, we use ``sample-hostname``
+as the hostname and ``27017`` as the port. Replace these values to point to
+your MongoDB instance.
+
+The last part of the connection string contains connection and authentication
+options as parameters. In the example below, we set two connection options:
+``retryWrites=true`` and ``w=majority``. For more information on connection
+options, skip to the :ref:`connection options <connection-options>` section.
+
+The code below shows how you can use the sample connection URI in a client to
+connect to MongoDB.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/srv.java
+   :language: java
+
+.. _connect-replica-set:
+
+Connect to a Replica Set
+------------------------
+
+A MongoDB replica set deployment is a group of connected instances that
+store the same set of data. This configuration of instances provides data
+redundancy and high data availability.
+
+To connect to a replica set deployment, specify the hostname (or IP address)
+and port numbers of one or more members of the replica set separated by commas.
+Use the  to specify the name
+of the replica set with which your application is connecting. When the
+replica set option is specified, the driver assumes that your
```

Comment:
Not sure what you mean here.  What is the assumption?  

I think what you're referring to is that:

`mongodb://<hostname:port>` will create a direct connection to a single server, even if that server is a member of a replica set.  whereas `mongodb://<hostname:port>/?replicaSet=rs0` will create a replica set connection, and discover all the members.  

If so, you should clarify